packages/visualizations/src/components/utils/ColorsLegend.svelte

```diff
@@ -114,4 +184,16 @@
     .legend-colors-color-box-palette:not(:last-child) {
         margin-right: 1px;
     }
+    /* Handle labels rotation */
+    .vertical-labels-container {
+        margin: 6px -6px;
+    }
+    .vertical-label {
+        transform: rotate(270deg);
+        height: var(--label-row-width);
+        min-width: var(--label-row-width);
+        line-height: var(--label-row-width);
+        margin: auto;
+        text-align: right;
+    }
```

Comment:
```suggestion
    .legend-colors-row-color-box-palette,
    .legend-colors-row-values-palette {
        …
        line-height: 1em; /* for easier alignement */
    }
    
    ...
    .label-container {
        flex: 1 1 auto;
        display: flex;
        justify-content: center;
        align-items: flex-start;
    }

    .label-container span {
        margin-left: calc(-50% - 1em);
    }

    .vertical-label {
        writing-mode: vertical-lr; /* ensure parent height */
        transform: rotate(180deg);
    }
```
So the idea here is to use writing mode vertical, so that that the height is computed at layout time (contrary to transform: rotate) and it keeps the width in the "natural" direction. Then we offset to have all labels on the segment separators.